### PR TITLE
Make sure to wait for the service before declaring events.

### LINKED ISSUE #472: 👩‍🌾 test_tutorial_parameter_events_async__rmw_connext_cpp regression on Windows Release
This test has failed in the last 5/6 most recent builds (since 2020-10-28), and it was passing consistently before that:

https://ci.ros2.org/view/nightly/job/nightly_win_rel/1740/testReport/(root)/projectroot/test_tutorial_parameter_events_async__rmw_connext_cpp/history/

<details><summary>Test failure output</summary>
<p>

```
-- run_test.py: extra environment variables:
 - RCL_ASSERT_RMW_ID_MATCHES=rmw_connext_cpp
 - RMW_IMPLEMENTATION=rmw_connext_cpp
-- run_test.py: invoking following command in 'C:\ci\ws\build\demo_nodes_cpp':
 - C:/Python38/python.exe -m launch_testing.launch_test C:/ci/ws/build/demo_nodes_cpp/test_parameter_events_async__rmw_connext_cpp_Release.py --junit-xml=C:/ci/ws/build/demo_nodes_cpp/test_results/demo_nodes_cpp/test_tutorial_parameter_events_async__rmw_connext_cpp.xunit.xml --package-name=demo_nodes_cpp
[INFO] [launch]: All log files can be found below C:\Users\ContainerAdministrator\.ros\log\2020-11-02-11-25-42-031390-c18fd2498ae4-5020
[INFO] [launch]: Default logging verbosity is set to INFO
test_processes_output (demo_nodes_cpp.TestExecutablesTutorial)
Test all processes output against expectations. ... [INFO] [test_executable_0-1]: process started with pid [7068]
[INFO] [python.exe-2]: process started with pid [16352]
[test_executable_0-1] COMMENDBeWriterService_assertRemoteReader:Discovered remote reader with GUID 0X10F777D,0X14380000,0X1000000,0X200C7 using a non-addressable locator.
[test_executable_0-1] This can occur if the transport is not installed and/or enabled in the local participant.
[test_executable_0-1] See https://community.rti.com/kb/what-does-cant-reach-locator-error-message-mean for additional info.
[test_executable_0-1] can't reach: locator: 16://5577:7D00:0000:0000:0000:0000:0000:0000:7662
[test_executable_0-1] COMMENDBeWriterService_assertRemoteReader:Discovered remote reader with GUID 0X10F777D,0XB8260000,0X1000000,0X200C7 using a non-addressable locator.
[test_executable_0-1] This can occur if the transport is not installed and/or enabled in the local participant.
[test_executable_0-1] See https://community.rti.com/kb/what-does-cant-reach-locator-error-message-mean for additional info.
[test_executable_0-1] can't reach: locator: 16://5577:7D00:0000:0000:0000:0000:0000:0000:7664
[test_executable_0-1] COMMENDSrWriterService_assertRemoteReader:Discovered remote reader with GUID 0X10F777D,0X14380000,0X1000000,0X3C7 using a non-addressable locator.
[test_executable_0-1] This can occur if the transport is not installed and/or enabled in the local participant.
[test_executable_0-1] See https://community.rti.com/kb/what-does-cant-reach-locator-error-message-mean for additional info.
[test_executable_0-1] can't reach: locator: 16://5577:7D00:0000:0000:0000:0000:0000:0000:7662
[test_executable_0-1] COMMENDSrWriterService_assertRemoteReader:Discovered remote reader with GUID 0X10F777D,0XB8260000,0X1000000,0X3C7 using a non-addressable locator.
[test_executable_0-1] This can occur if the transport is not installed and/or enabled in the local participant.
[test_executable_0-1] See https://community.rti.com/kb/what-does-cant-reach-locator-error-message-mean for additional info.
[test_executable_0-1] can't reach: locator: 16://5577:7D00:0000:0000:0000:0000:0000:0000:7664
[test_executable_0-1] COMMENDSrWriterService_assertRemoteReader:Discovered remote reader with GUID 0X10F777D,0X14380000,0X1000000,0X4C7 using a non-addressable locator.
[test_executable_0-1] This can occur if the transport is not installed and/or enabled in the local participant.
[test_executable_0-1] See https://community.rti.com/kb/what-does-cant-reach-locator-error-message-mean for additional info.
[test_executable_0-1] can't reach: locator: 16://5577:7D00:0000:0000:0000:0000:0000:0000:7662
[test_executable_0-1] COMMENDSrWriterService_assertRemoteReader:Discovered remote reader with GUID 0X10F777D,0XB8260000,0X1000000,0X4C7 using a non-addressable locator.
[test_executable_0-1] This can occur if the transport is not installed and/or enabled in the local participant.
[test_executable_0-1] See https://community.rti.com/kb/what-does-cant-reach-locator-error-message-mean for additional info.
[test_executable_0-1] can't reach: locator: 16://5577:7D00:0000:0000:0000:0000:0000:0000:7664
[test_executable_0-1] 
[test_executable_0-1] Parameter event:
[test_executable_0-1]  new parameters:
[test_executable_0-1]  changed parameters:
[test_executable_0-1]   foo
[test_executable_0-1]  deleted parameters:
[test_executable_0-1] 
[test_executable_0-1] 
[test_executable_0-1] Parameter event:
[test_executable_0-1]  new parameters:
[test_executable_0-1]  changed parameters:
[test_executable_0-1]   bar
[test_executable_0-1]  deleted parameters:
[test_executable_0-1] 
[test_executable_0-1] 
[test_executable_0-1] Parameter event:
[test_executable_0-1]  new parameters:
[test_executable_0-1]  changed parameters:
[test_executable_0-1]   baz
[test_executable_0-1]  deleted parameters:
[test_executable_0-1] 
[test_executable_0-1] 
[test_executable_0-1] Parameter event:
[test_executable_0-1]  new parameters:
[test_executable_0-1]  changed parameters:
[test_executable_0-1]   foobar
[test_executable_0-1]  deleted parameters:
[test_executable_0-1] 
[test_executable_0-1] 
[test_executable_0-1] Parameter event:
[test_executable_0-1]  new parameters:
[test_executable_0-1]  changed parameters:
[test_executable_0-1]   foo
[test_executable_0-1]  deleted parameters:
[test_executable_0-1] 
[test_executable_0-1] 
[test_executable_0-1] Parameter event:
[test_executable_0-1]  new parameters:
[test_executable_0-1]  changed parameters:
[test_executable_0-1]   bar
[test_executable_0-1]  deleted parameters:
[test_executable_0-1] 
[INFO] [test_executable_0-1]: process has finished cleanly [pid 7068]
[WARNING] [python.exe-2]: 'SIGINT' sent to process[python.exe-2] not supported on Windows, escalating to 'SIGTERM'
[INFO] [python.exe-2]: sending signal 'SIGTERM' to process[python.exe-2]
[ERROR] [python.exe-2]: process has died [pid 16352, exit code 1, cmd 'C:\Python38\python.exe -c 
try:
    while True:
        pass
except KeyboardInterrupt:
    pass
'].
FAIL

======================================================================
FAIL: test_processes_output (demo_nodes_cpp.TestExecutablesTutorial)
Test all processes output against expectations.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\ci\ws\build\demo_nodes_cpp\test_parameter_events_async__rmw_connext_cpp_Release.py", line 55, in test_processes_output
    proc_output.assertWaitFor(
  File "C:\ci\ws\install\Lib\site-packages\launch_testing\io_handler.py", line 146, in assertWaitFor
    assert success, 'Waiting for output timed out'
AssertionError: Waiting for output timed out

```

</p>
</details>

### PR DESCRIPTION
**Pull request by clalancette**
Otherwise, we can sometimes miss the initial event when the
parameters are first declared.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

This should fix #472, and hence a flaky test on the buildfarm.

### CODE REVIEW

**Comment by clalancette** [comment]
CI (only demo_nodes_cpp, retest-until-fail 10):
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12966/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7916/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10684/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12916/)

File: demo_nodes_cpp/test/test_executables_tutorial.py.in [hunk 1]
```diff
@@ -17,7 +17,6 @@ import launch_testing_ros
 
 def generate_test_description():
     env = os.environ.copy()
-    env['OSPL_VERBOSITY'] = '8'  # 8 = OS_NONE
```
Comment (hidmic) [inline demo_nodes_cpp/test/test_executables_tutorial.py.in:20]: @clalancette unrelated?

**Review by hidmic — APPROVED** [review]
That extra snippet is... annoying. Correct. But annoying.

File: demo_nodes_cpp/test/test_executables_tutorial.py.in (hunk 1 above)
Comment (clalancette) [reply demo_nodes_cpp/test/test_executables_tutorial.py.in:20]: Yeah, I just noticed it while I was in here.  It's to support Opensplice stuff, and we don't need it anymore.  I can remove it from this PR if you don't think it should be in here.

**Comment by clalancette** [comment]
> This extra loop is... annoying. Correct. But annoying.

Yeah, I agree.  At the very least, having it be correct here in the demos means that users looking at it will have some idea that they need to do it.  But it is really not that obvious that you need to do it.

**Review by wjwwood — APPROVED** [review]

File: demo_nodes_cpp/test/test_executables_tutorial.py.in (hunk 1 above)
Comment (clalancette) [reply demo_nodes_cpp/test/test_executables_tutorial.py.in]: I ended up not including this in the patch I just pushed.  That just keeps this PR focused.  I may do a later cleanup where I remove this from a bunch of places.

**Comment by clalancette** [comment]
I'm going to go ahead and merge this.  Thanks for the reviews!

**Pull request merged**